src/devices/Mfrc522/Mfrc522.cs

```diff
@@ -0,0 +1,959 @@
+        /// Constructor for MFRC5222 with Serial Port interface.
+        /// </summary>
+        /// <param name="serialPort">A Serial Port, default speed is 9600 baud, no parity, 1 bit stop.</param>
+        /// <param name="pinReset">A reset pin for the hardware reset.</param>
+        /// <param name="gpioController">A GpioController for the hardware reset.</param>
+        /// <param name="shouldDispose">True to dispose the GpioController.</param>
+        public MfRc522(SerialPort serialPort, int pinReset = -1, GpioController? gpioController = null, bool shouldDispose = true)
+        {
+            _serialPort = serialPort;
+            _serialPort.ReadTimeout = 1000;
+            _serialPort.WriteTimeout = 1000;
+            if (!_serialPort.IsOpen)
+            {
+                _serialPort.Open();
+            }
+
+            _pinReset = pinReset;
+
+            HardReset(gpioController, shouldDispose);
+            SetDefaultValues();
+        }
+
+        private void SetDefaultValues()
+        {
+            // Switch off the crypto for Mifare card in case it's on
+            ClearRegisterBit(Register.Status2, (byte)Status2.MFCrypto1On);
+            // Set Timer for Timeout, see documentation on those registers to understand the values
+            WriteRegister(Register.TMode, (byte)TMode.TAutoRestart);
+            WriteRegister(Register.TPrescaler, 0xA9);
+            WriteRegister(Register.TReloadHigh, 0x06);
+            WriteRegister(Register.TReloadLow, 0xE8);
+
+            // forces a 100 % ASK modulation independent of the ModGsPReg register setting
+            WriteRegister(Register.TxAsk, 0x40);
+
+            // Set CRC to 0x6363 (iso 14443-3 6.1.6)
+            WriteRegister(Register.Mode, (byte)(Mode.TxWaitRF | Mode.PolMFinHigh) | (byte)ModeCrc.Preset6363h);
+            // Switching on the antenna
+            Enabled = true;
+        }
+
+        private void HardReset(GpioController? gpioController, bool shouldDispose)
+        {
+            if (_pinReset >= 0)
+            {
+                _shouldDispose = shouldDispose || gpioController == null;
+                _controller = gpioController ?? new GpioController();
+                _controller.OpenPin(_pinReset, PinMode.Output);
+                _controller.Write(_pinReset, PinValue.Low);
+                // 100 nano seconds at least, take some margin
+                Thread.Sleep(1);
+                _controller.Write(_pinReset, PinValue.High);
+                // 37.7 milliseconds according to documentation
+                Thread.Sleep(38);
+            }
+        }
+
+        #endregion
+
+        #region Properties and functions
+
+        /// <summary>
+        /// Get or Set the gain.
+        /// </summary>
+        public Gain Gain
+        {
+            get => (Gain)(ReadRegister(Register.RFCfg) & (byte)Gain.G48dB);
+            set
+            {
+                WriteRegister(Register.RFCfg, (byte)value);
+            }
```

Comment:
```suggestion
            set => WriteRegister(Register.RFCfg, (byte)value);
```

for consistency with get